app/models/order.rb

```diff
@@ -1,7 +1,9 @@
 class Order < ApplicationRecord
+  enum status: {processing: 0, shipped: 1, completed: 2, canceled: 3}
+
   belongs_to :user, dependent: :destroy
-  has_many :order_product, dependent: :destroy
+  has_many :order_products, dependent: :destroy
+  has_many :products, through: :order_products
 
-  validates :total, presence: true
-  validates :data, presence: true
+  scope :order_by_newest, ->{order(:created_at)}
```

Comment:
```suggestion
  scope :order_by_newest, ->{order :created_at}
```
cơ mà chị nghĩ nên sắp xếp theo status chứ nhỉ

Reply:
Dạ. Rứa để em sắp xếp lại :D